File: jumpgame2.cpp
class Solution {
public:
     int jumpGameII(vector<int>&jumps){
        int answer = 0;
         int left = 0, right = 0;


         int n = jumps.size();
         while(right < n){
             int farthest = 0;
              for(int i=left;i<=right;i++){
                farthest = max(farthest, i + nums[i]);
             }

             left = right + 1;
             right = farthest;
             answer++;
         }

         

         return answer;

     }
};

File: jumpGame2.py
class Solution:
    def jumpGameII(self, jumps):
        answer = 0
        left = 0
        right = 0

        n = len(jumps)

        while right < n:
            farthest = 0
            for i in range(left, right+1):
                farthest = max(farthest, i + jumps[i])
            left = right + 1
            right = farthest
            answer += 1

        return answer
